### PR TITLE
JetStream API clean-up

### PR DESCRIPTION
**Pull request by mtmk**
* Removed opinionated method overloads for create consumer and stream.
* Removed publish overload for sending empty messages. Doesn't feel like a common operation in JS. Removed to reduce the API footprint.
* A few format fixes.
* A couple of potential test flapper timeout fixes.

/cc @Jarema

### CODE REVIEW

**Comment by caleblloyd** [comment]
I do miss the simplicity, it's no longer clear what settings are actually required.  Would it make sense to provide some static helpers/extensions

```
// helpers
static StreamConfig(string streamName, IEnumerable<string> subjects) StreamCreateRequest
static ConsumerConfig(string streamName, string consumerName) ConsumerCreateRequest

// extensions
static ConsumerConfig(INatsJetStream streamName, string consumerName) ConsumerCreateRequest
```

That way if I want an R3 stream I could do

```
var sc = new StreamConfig("my-stream", []{"my-stream"}) with {Replicas = 3}
```

Much simpler in my opinion than having to look into the detials of the record to discover that `Name` and `Subjects` are required for me to actually get a working stream

Also - not sure how often people create Ephemeral consumers explicitly.  Might be worth making the helpers explicit for `DurableConsumerConfig` and `EphemeralConsumerConfig`.  Ephemeral Consumers in that case could have random names.

It takes some deep knowledge to know that you must supply `Name` and `DurableName`

**Comment by caleblloyd** [comment]
Any interest in shortening the models `ConsumerConfiguration` => `ConsumerConfig` and `StreamConfiguration` => `StreamConfig`...  And any other model with `Configuration` => `Config`...  It is a long word :smile:

**Review by caleblloyd — APPROVED** [review]
LGTM

**Comment by Jarema** [comment]
I like the change to `Config`!

File: sandbox/Example.JetStream.PullConsumer/Program.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 var js = new NatsJSContext(nats);
 
-var consumer = await js.CreateConsumerAsync(new ConsumerCreateRequest { StreamName = "s1", Config = new ConsumerConfiguration { Name = "c1", DurableName = "c1", AckPolicy = ConsumerConfigurationAckPolicy.@explicit } });
+var consumer = await js.CreateConsumerAsync("s1", new ConsumerConfiguration { Name = "c1", DurableName = "c1", AckPolicy = ConsumerConfigurationAckPolicy.@explicit });
```
Comment (oising) [inline sandbox/Example.JetStream.PullConsumer/Program.cs]: Framework guidelines dictate that enum names should be Pascal cased -- then you could avoid the ugly at escape on explicit.

Comment (mtmk) [reply sandbox/Example.JetStream.PullConsumer/Program.cs]: Unfortunately I had an issue with the JSON serializer not using the attributed name then had to make it same case as the JSON property to make it work.

Comment (oising) [reply sandbox/Example.JetStream.PullConsumer/Program.cs]: For S.T.Json, the way I do this [net8] is:

```csharp
builder.Services
   .AddJsonOptions(
        options =>
        {
            options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter<PassportIssuingSettings>(JsonNamingPolicy.CamelCase));
            options.JsonSerializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
            
            options.JsonSerializerOptions.TypeInfoResolver = PassportJsonSerializationContext.Default;
 ```

Comment (oising) [reply sandbox/Example.JetStream.PullConsumer/Program.cs]: PassportJsonSerializationContext is my source generator context for my structures. I'd imagine you guys are using that code path too?

Comment (oising) [reply sandbox/Example.JetStream.PullConsumer/Program.cs]: Here's the partial context I declared: 

```csharp
[JsonSerializable(typeof(Passport))]
[JsonSerializable(typeof(PassportIssuingSettings))]
[JsonSerializable(typeof(Visa))]
[JsonSerializable(typeof(ValidationProblemDetails))]
[JsonSourceGenerationOptions(PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase)]
public partial class PassportJsonSerializationContext : JsonSerializerContext
{
}
```
This is for AOT ready serialization etc.

Comment (oising) [reply sandbox/Example.JetStream.PullConsumer/Program.cs]: Tbh, some of my declarations are probably redundant - anything in the AddJsonOptions setup can probably be applied to the context instead. I went through an evolution making things work from net6 to net8 :/

Comment (mtmk) [reply sandbox/Example.JetStream.PullConsumer/Program.cs]: Yes we have a context, that attribute solution looks promising. Do you still have to set the option?

Comment (oising) [reply sandbox/Example.JetStream.PullConsumer/Program.cs]: I think the option is only for when you don't use a context. I started out that way, then added a context later but never took away the option. I _assume_ the attributes on the context are enough :)

Comment (oising) [reply sandbox/Example.JetStream.PullConsumer/Program.cs]: Oh, so you can pass in options director to the context ctor too:

```csharp
_jsonSerializationContext = new PassportJsonSerializationContext(
    new JsonSerializerOptions
    {
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
        Converters = {  }
    });
 ```
 
 Then you don't need the entry point AddJsonOptions fluff.

Comment (mtmk) [reply sandbox/Example.JetStream.PullConsumer/Program.cs]: hmmm. it doesn't seem to work (see example https://github.com/mtmk/example-json-enum)

Need to implement a convertor I think.

**Pull request merged**